jdaviz/utils.py

```diff
@@ -250,7 +250,17 @@ def is_wcs_only(layer):
         state = layer
     else:
         raise NotImplementedError
-    return getattr(state, 'meta', {}).get(_wcs_only_label, False)
+
+    is_subset = hasattr(layer, 'subset_state') or hasattr(state, 'subset_state')
+
+    return (
+        # WCS-only layers have a metadata label:
+        getattr(state, 'meta', {}).get(_wcs_only_label, False) and
+
+        # the above logic doesn't exclude 100% of subsets, so here we
+        # ensure that no subsets make it through the filter:
+        not is_subset
+    )
```

Comment:
wait, so `subset.state.meta['_WCS_ONLY']` is sometimes set _and_ True? 🤔

Reply:
I tried to simplify this whole function, but there were corner cases that got messy. The layers that get passed through `LayerSelect` can have the following attributes: `data`, `layer`, or `meta`:
https://github.com/spacetelescope/jdaviz/blob/23b3edd44e8e40426d3035a100241bd65bd93cab/jdaviz/utils.py#L244-L252
In the case where the layer is a subset, `layer.data` corresponds to the subset's parent Data, which itself may have a `meta['_WCS_ONLY']`. That's where the second check is necessary.

Reply:
I'm experimenting right now to see if this works without the second `elif`. 🤞🏻

Reply:
Phew, much simplified in b13c3bc.